src/sources/Dataset.js

```diff
@@ -19,6 +19,8 @@ export default class Dataset extends BaseWindshaft {
      * If your server is not hosted by CARTO you must add a third parameter that includes the serverURL. This can be done {@link carto.setDefaultConfig|setting the default config} in the carto object or providing a `config`
      * object with your serverURL.
      *
+     * The combination of different type of geometries on the same source is unsupported. Valid geometry types are `points`, `lines` and `polygons`.
```

Comment:
I would use `not supported` instead of `unsupported` because maybe someone thinks that it was supported before or it can be supported in the future.